src/spdx_matcher/__init__.py

```diff
@@ -206,142 +209,203 @@ def normalize(license_text, remove_sections=REMOVE_FINGERPRINT):
+
+    :param cache_file: The path of the cache file
+    :param match_cache: The dict of match_cache
+    """
+    cache_data_match_cost = {"licenses": {}, "exceptions": {}}
+    with open(cache_file, "r", encoding="utf-8") as cf:
+        cache_data = json.load(cf)
+        cache_data_match_cost["licenses"] = {
+            license_id: license_data["matchCost"]
+            for license_id, license_data in cache_data["licenses"].items()
+        }
+        cache_data_match_cost["exceptions"] = {
+            license_exception_id: exception_data["matchCost"]
+            for license_exception_id, exception_data in cache_data["exceptions"].items()
+        }
+
+    for license_id, license_data in match_cache["licenses"].items():
+        if license_id in cache_data_match_cost["licenses"]:
+            license_data["matchCost"] = cache_data_match_cost["licenses"][license_id]
+    for license_exception_id, exception_data in match_cache["exceptions"].items():
+        if license_exception_id in cache_data_match_cost["exceptions"]:
+            exception_data["matchCost"] = cache_data_match_cost["exceptions"][license_exception_id]
+
+
+def cache_builder(change_match_cost: bool | None = True) -> None:
+    """
+    Builds the cache file base on spdx licenses and exceptions json files
+
+    :param change_match_cost: Whether change attribute `matchCost` in cache data file, use to reduce the diff
+        of git commit, Will use the realtime match cost for current runtime if True, otherwise will use
+        previous match cost in the cache data file. Default is True.
+    """
+    match_cache = {"licenses": {}, "exceptions": {}}
+    base_url = "https://spdx.org/licenses"
+
+    # Fetch licenses.json
+    licenses_data = _fetch_json(f"{base_url}/licenses.json")
+    exceptions_data = _fetch_json(f"{base_url}/exceptions.json")
+
+    # Process licenses and exceptions using ThreadPoolExecutor
+    licenses = []
+    exceptions = []
+    with ThreadPoolExecutor(max_workers=DEFAULT_CACHE_THREAD_POOL_WORKERS) as executor:
+        # Process licenses concurrently
+        for license in licenses_data["licenses"]:
+            licenses.append(executor.submit(_process_license, license))
+
+        # Process exceptions concurrently
+        for exception in exceptions_data["exceptions"]:
+            exceptions.append(executor.submit(_process_exception, exception))
+
+    wait(licenses)
+    wait(exceptions)
+    for license in licenses:
+        match_cache["licenses"].update(license.result())
+    for exception in exceptions:
+        match_cache["exceptions"].update(exception.result())
+
+    cache_file = os.getenv("SPDX_MATCHER_CACHE_FILE", DEFAULT_CACHE_PATH)
+    if not change_match_cost:
+        logger.info("Not changing the match cost in the cache data file, due to parameter "
+                    "`change_match_cost` is False")
+        _merge_cache_data_match_cost(cache_file, match_cache)
```

Comment:
Hi Mike, I add `change_match_cost` to control whether we should overwrite cache data with the latest `matchCost` or not. I found out we use `matchCost` to save the time of license match when we use it. And In most of case we do not need to change it often, but event time we build cache `matchCost` have different value and it cost a lot of git diff, so I add this parameter to control whether is should be overwrite or not.

And the default value in __init__.py call is True, which mean we do not change the behavior